tests/TestCase.php

```diff
@@ -46,13 +46,77 @@ abstract class TestCase extends PHPTestCase
 
     protected $rootDir;
 
+    private $catchAllErrorHandler;
+
+    function __construct() {
+        parent::__construct();
+
+        // PHP does not implement setting a class property to an anonymous function,
+        // so we have to do it in the constructor.
+        $this->catchAllErrorHandler = function ($typeNumber, $message, $file, $lineNumber) {
+            $this->fail(
+                sprintf('%s: "%s" in %s:%d',
+                    $this->getErrorType($typeNumber),
+                    $message,
+                    $file,
+                    $lineNumber
+                )
+            );
+        };
+    }
+
     public function setUp()
     {
         parent::setUp();
 
         $this->rootDir = __DIR__.'/..';
     }
 
+    public function tearDown() {
+        // if we changed the error handler using catchAllErrors(), reset it now
+        $currentErrorHandler = set_error_handler('var_dump');
+        restore_error_handler();
+        if($currentErrorHandler === [$this, 'catchAllErrorHandler']) {
+            restore_error_handler();
+        }
+    }
+
+    /**
+     * This temporarily changes the PHP-internal error handler
+     * in order to allow catching errors of type E_WARNING, E_NOTICE etc.,
+     * which are not catchable via a try/catch statement.
+     * It will fail the current test from which it is run,
+     * giving a descriptive message.
+     *
+     * This can come in handy to make tests for making sure that such
+     * errors are not triggered by the code.
+     */
+    protected function catchAllErrors() {
+        set_error_handler($this->catchAllErrorHandler);
+    }
+
+    protected function getErrorType($typeNumber) {
+        $errorConstants = [
+            1 => 'E_ERROR',
+            2 => 'E_WARNING',
+            4 => 'E_PARSE',
+            8 => 'E_NOTICE',
+            16 => 'E_CORE_ERROR',
+            32 => 'E_CORE_WARNING',
+            64 => 'E_COMPILE_ERROR',
+            128 => 'E_COMPILE_WARNING',
+            256 => 'E_USER_ERROR',
+            512 => 'E_USER_WARNING',
+            1024 => 'E_USER_NOTICE',
+            2048 => 'E_STRICT',
+            4096 => 'E_RECOVERABLE_ERROR',
+            8192 => 'E_DEPRECATED',
+            16384 => 'E_USER_DEPRECATED',
+            32767 => 'E_ALL'
```

Comment:
I think that is a little bit out of scope. Is it really necessary to create a list of PHP error types? Why not just print the error message and available meta data.

Reply:
Yeah, it might be a bit over the top... I thought it could be helpful to see the type of error before the error message in a human-readable way when writing the tests. But if you think it's too much, I can remove it! Please let me know again.

Reply:
Please remove it here. But it could be worth looking into it again in another PR.